src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt

```diff
@@ -0,0 +1,126 @@
+package com.bridgecrew.analytics
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.diagnostic.logger
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.concurrent.TimeUnit
+
+@Service
+class AnalyticsService {
+    private val LOG = logger<AnalyticsService>()
+
+    private lateinit var fullScanData: FullScanData
+    private var fullScanNumber = 0
+
+    fun fullScanButtonWasPressed() {
+        fullScanNumber += 1
+        LOG.info("Prisma Plugin Analytics - scan #${fullScanNumber} - full scan button was pressed")
+        fullScanData = FullScanData(fullScanNumber)
+        fullScanData.fullScanButtonPressedTime = Date()
```

Comment:
I would put that on the first line, currently not much difference but if the logic gets more complicated you want the time to reflect the click